### PR TITLE
Ensure timeout value in await_install less than MAX_JOB_TIME

### PR DESCRIPTION
**Pull request by lemon-suse**
We should ensure timeout value in await_install less than MAX_JOB_TIME.

- Related ticket: https://progress.opensuse.org/issues/64228
- Needles: N/A
- Verification run: http://openqa.nue.suse.com/tests/3968672

### CODE REVIEW

File: tests/installation/await_install.pm [hunk 1]
```diff
@@ -123,6 +123,8 @@ sub run {
     if (check_var('ARCH', 's390x')) {
         push(@tags, 'additional-packages');
     }
+    # For poo#64228, we need ensure the timeout value less than the MAX_JOB_TIME
+    $timeout = get_var('MAX_JOB_TIME') - 1000 if ($timeout > get_var('MAX_JOB_TIME'));
```
Comment (okurz) [inline tests/installation/await_install.pm]: I would prefer a hard failure early in the tests if MAX_JOB_TIME is not high enough instead of silently lowering the timeout. This can be very hard to follow for people why above timeout is set to big values and then we silently set it lower values here. How about:

```perl
my $max_job_time_bound = get_required_var('MAX_JOB_TIME') - 1000;
die "Computed timeout '$timeout' exceeds max_job_time_bound '$max_job_time_bound', consider decreasing '$timeout' or increasing 'MAX_JOB_TIME'" if $timeout > $max_job_time_bound;
```

This way it will be very explicit what needs to be changed.

Comment (lemon-suse) [reply tests/installation/await_install.pm]: Understand it, I updated code following your suggestion. I will run a verify log for it.

Comment (okurz) [reply tests/installation/await_install.pm]: looks good, thx. Waiting for the verification run.

**Comment by lemon-suse** [comment]
Updated the verify log. http://openqa.nue.suse.com/tests/3978114#step/await_install/2

**Comment by okurz** [comment]
great. thx. Again, not so much details in git commit message but only in PR description but in this case I consider the git subject line good enough :)

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/1200247#step/await_install/2

**Comment by okurz** [comment]
@lemon-suse reverted PR in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9758 . Problem seems to be that MAX_JOB_TIME is by default not defined.

**Comment by lemon-suse** [comment]
> @lemon-suse reverted PR in #9758 . Problem seems to be that MAX_JOB_TIME is by default not defined. Maybe you can check where the value comes from. If we don't have any better idea than we can duplicate the default value and replace `get_required_var('MAX_JOB_TIME')` with `get_var('MAX_JOB_TIME', 7200)`. But I thought maybe it's better if we don't die but just `record_info` for a certain time, say 2 weeks, and only then use die again to enforce the check after people had a chance to react on the notice.

I'm not sure how to do this in openQA, "`record_info` for a certain time, say 2 weeks, and only then use die again to enforce the check after people had a chance to react on the notice."

**Comment by okurz** [comment]
Not in openQA, have a PR with record_info instead of die and prepare another commit that we only merge after some time :)

**Comment by lemon-suse** [comment]
> Not in openQA, have a PR with record_info instead of die and prepare another commit that we only merge after some time :)

New PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9760
